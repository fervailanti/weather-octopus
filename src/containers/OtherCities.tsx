import React, { FC, ReactElement, useContext } from 'react'
import Chip from '../components/Chip'
import { cities } from '../helpers/constans/cities'
import { resolveCountry } from '../helpers/methods/resolveCountry'
import { CoordType } from '../helpers/types'
import { PinnedCityContext } from '../providers/PinnedCity'

const OtherCities: FC = (): ReactElement => {
  const { pinnedCity, setPinnedCity } = useContext(PinnedCityContext)

  const cityAction = (id: number, coord: CoordType) => {
    setPinnedCity({ type: 'id', id })
    setPinnedCity({ type: 'coord', coord })
    const element: any = document.querySelector('#pinned-city-forecast')
    const bodyRect = document.body.getBoundingClientRect().top
    const elementRect = element.getBoundingClientRect().top
    const offsetPosition = elementRect - bodyRect - 60
    window.scrollTo({ top: offsetPosition, behavior: 'smooth' })
  }

  return (
    <div className="flex flex-col items-center md:mt-4 xl:mt-8">
      <p className="text-white font-thin text-xl my-3">Other cities you may also like:</p>
      <div className="flex md:flex-row flex-wrap justify-center mt-0">
        {cities
          .filter((e) => e.id !== pinnedCity.id)
          .map(({ id, name, country, image, coord }) => (
            <Chip
              key={id}
              title={name}
              subtitle={resolveCountry(country)}
              image={image}
              action={() => cityAction(id, coord)}
            />
          ))}
      </div>
    </div>
  )
}

export default OtherCities
