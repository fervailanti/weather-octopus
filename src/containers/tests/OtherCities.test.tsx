import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import OtherCities from '../OtherCities'

describe('CurrentWeather Container', () => {
  // Complex to test because of it's nature.
  // At least, it should render without explode and show cities correctly.
  it('Should render correctly and show cities correctly', () => {
    const otherCities = render(<OtherCities />)
    expect(otherCities.container).toMatchSnapshot()
    expect(otherCities.container).toHaveTextContent('New York')
    expect(otherCities.container).toHaveTextContent('Paris')
    expect(otherCities.container).toHaveTextContent('Rome')
    expect(otherCities.container).toHaveTextContent('Tokyo')
    expect(otherCities.container).toHaveTextContent('Berlin')
  })
})
