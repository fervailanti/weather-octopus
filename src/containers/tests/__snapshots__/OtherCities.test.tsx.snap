// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentWeather Container Should render correctly and show cities correctly 1`] = `
<div>
  <div
    class="flex flex-col items-center md:mt-4 xl:mt-8"
  >
    <p
      class="text-white font-thin text-xl my-3"
    >
      Other cities you may also like:
    </p>
    <div
      class="flex md:flex-row flex-wrap justify-center mt-0"
    >
      <div
        class="bg-gray-400 rounded-lg shadow-xl max-w-xl md:w-48 my-2 md:mx-2 xl:mx-4 overflow-hidden lg:cursor-pointer transform lg:hover:scale-105 lg:hover:bg-gray-300 transition duration-150 ease-in-out active:opacity-50"
      >
        <img
          alt="alt"
          class="object-cover w-full h-24"
          src="new_york.jpg"
        />
        <div
          class="p-3"
        >
          <h2
            class="text-gray-900"
          >
            <span
              class="font-semibold text-xl"
            >
              New York
            </span>
            <span
              class="text-sm"
            >
              , 
              USA
            </span>
          </h2>
        </div>
      </div>
      <div
        class="bg-gray-400 rounded-lg shadow-xl max-w-xl md:w-48 my-2 md:mx-2 xl:mx-4 overflow-hidden lg:cursor-pointer transform lg:hover:scale-105 lg:hover:bg-gray-300 transition duration-150 ease-in-out active:opacity-50"
      >
        <img
          alt="alt"
          class="object-cover w-full h-24"
          src="paris.jpg"
        />
        <div
          class="p-3"
        >
          <h2
            class="text-gray-900"
          >
            <span
              class="font-semibold text-xl"
            >
              Paris
            </span>
            <span
              class="text-sm"
            >
              , 
              France
            </span>
          </h2>
        </div>
      </div>
      <div
        class="bg-gray-400 rounded-lg shadow-xl max-w-xl md:w-48 my-2 md:mx-2 xl:mx-4 overflow-hidden lg:cursor-pointer transform lg:hover:scale-105 lg:hover:bg-gray-300 transition duration-150 ease-in-out active:opacity-50"
      >
        <img
          alt="alt"
          class="object-cover w-full h-24"
          src="rome.jpg"
        />
        <div
          class="p-3"
        >
          <h2
            class="text-gray-900"
          >
            <span
              class="font-semibold text-xl"
            >
              Rome
            </span>
            <span
              class="text-sm"
            >
              , 
              Italy
            </span>
          </h2>
        </div>
      </div>
      <div
        class="bg-gray-400 rounded-lg shadow-xl max-w-xl md:w-48 my-2 md:mx-2 xl:mx-4 overflow-hidden lg:cursor-pointer transform lg:hover:scale-105 lg:hover:bg-gray-300 transition duration-150 ease-in-out active:opacity-50"
      >
        <img
          alt="alt"
          class="object-cover w-full h-24"
          src="tokyo.jpg"
        />
        <div
          class="p-3"
        >
          <h2
            class="text-gray-900"
          >
            <span
              class="font-semibold text-xl"
            >
              Tokyo
            </span>
            <span
              class="text-sm"
            >
              , 
              Japan
            </span>
          </h2>
        </div>
      </div>
      <div
        class="bg-gray-400 rounded-lg shadow-xl max-w-xl md:w-48 my-2 md:mx-2 xl:mx-4 overflow-hidden lg:cursor-pointer transform lg:hover:scale-105 lg:hover:bg-gray-300 transition duration-150 ease-in-out active:opacity-50"
      >
        <img
          alt="alt"
          class="object-cover w-full h-24"
          src="berlin.jpg"
        />
        <div
          class="p-3"
        >
          <h2
            class="text-gray-900"
          >
            <span
              class="font-semibold text-xl"
            >
              Berlin
            </span>
            <span
              class="text-sm"
            >
              , 
              Germany
            </span>
          </h2>
        </div>
      </div>
    </div>
  </div>
</div>
`;
